src/daemon/rpmostree-sysroot-upgrader.c

```diff
@@ -457,10 +413,15 @@ void
 rpmostree_sysroot_upgrader_set_origin_override (RpmOstreeSysrootUpgrader *self,
                                                 const char *override_commit)
 {
+  g_autoptr(GKeyFile) new_origin = rpmostree_origin_dup_keyfile (self->origin);
+
   if (override_commit != NULL)
-    g_key_file_set_string (self->origin, "origin", "override-commit", override_commit);
+    g_key_file_set_string (new_origin, "origin", "override-commit", override_commit);
   else
-    g_key_file_remove_key (self->origin, "origin", "override-commit", NULL);
+    g_key_file_remove_key (new_origin, "origin", "override-commit", NULL);
+
+  if (!parse_origin_keyfile (self, new_origin, NULL, NULL))
+    g_assert_not_reached ();
 
   /* just update self manually rather than re-parsing the whole thing */
```

Comment:
Avoiding issues like this is another good reason for absorbing all the fields in `RpmOstreeOrigin`.